core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -953,6 +957,16 @@ class ReplicaManager(val config: KafkaConfig,
         }
       }
     }
+
+    requestedTopicSet.foreach { case topic =>
+      brokerTopicStats.topicStats(topic).totalProduceRequestRate.mark()
+    }
```

Comment:
nit: maybe something like
`requestedTopics.map(brokerTopicStats.topicStats).foreach(_.totalProduceRequestRate.mark())`
flows a bit better.

Reply:
The downside of using 'map' is that it will create an intermediate collection.

Reply:
this is a good point. In that case it may be fine to keep the update code as is. Maybe change:

`case topic =>` to just `topic =>` as i dont think the case is necessary for iterating over a set.

Reply:
all // in short, just replace `case topic =>` to `topic =>`; do I understand correctly?